### DISCUSSION TITLE
Customization best practices?

### DISCUSSION THREAD

**Discussion by LisaFC** [google/docsy#552] [CSS and other site hacks]
Hi hi. We've been excited to see all the different ways people have customized Docsy for their sites, and would like to put together a best practices doc. Things like considering the potential maintenance overhead, is anyone else interested in the feature/customization (check the issues and PRs!).....

Does anyone have any more suggestions/examples to get us started?

**Comment by marshalc** [comment]
I could certainly benefit from a best practice doc. I'm in the process of rebuilding our customisations to the core docsy setup, and one of those key changes is that we have a corporate colour set, and a range of primary (and all supporting) colour options - which we then allow our various teams to pick from for their documentation builds.

Thus far this has meant making an inline CSS include change to the partials/head-css.html to allow the config to drive an include for the selected colour theme, and then having all the possible options defined in SCSS to support that. I'm left wondering if there's a better way to do this...?

Our first attempt (which has worked for the past year or so) is viewable at <https://github.com/ouhft/docsy/tree/ouhft-broken>. The rebuild of this is going to be at <https://github.com/ouhft/docsy/tree/ouhft> (WIP as I write)

**Reply by marshalc** [reply]
More detail on our theme choosing solution at https://github.com/google/docsy/issues/1845#issuecomment-1943707651
